src/coreclr/src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -3145,6 +3145,12 @@ HRESULT ProfToEEInterfaceImpl::GetAppDomainStaticAddress(ClassID classId,
         return CORPROF_E_DATAINCOMPLETE;
     }
 
+    if (typeHandle.GetModule()->GetLoaderAllocator() == NULL ||
```

Comment:
@davmason I assume the runtime is suspended when we run this code, correct?

Reply:
We don't enforce that the runtime is suspended, but it would be dangerous even without this fix. This API returns a pointer to a managed static field, so with collectible contexts the object could disappear on you if you call it at arbitrary times. The expected code sequence is that this is called in the GC finished profiler callback, which happens before the runtime is resumed.

Reply:
Great, thanks.